kunta-api-server/src/test/java/fi/otavanopisto/kuntaapi/test/AbstractTest.java

```diff
@@ -0,0 +1,180 @@
+  }
+
+  protected long insertSystemSetting(String key, String value) {
+    return executeInsert("insert into SystemSetting (settingKey, value) values (?, ?)", key, value);
+  }
+  
+  protected void deleteOrganizationSetting(String organizationId, String key) {
+    executeDelete("delete from OrganizationSetting where settingKey = ? and organizationKuntaApiId = ?", key, organizationId);
+  }
+
+  protected void deleteSystemSetting(String key) {
+    executeDelete("delete from SystemSetting where settingKey = ?", key);
+  }
+
+  protected long createIdentifier(String kuntaApiId, String source, String sourceId, String type) {
+    return executeInsert("insert into Identifier (kuntaApiId, source, sourceId, type) values (?, ?, ?, ?)", 
+      kuntaApiId, source, sourceId, type);
+  }
+
+  protected void deleteIndentifier(String key) {
+    executeDelete("delete from Identifier where kuntaApiId = ?", key);
+  }
+  
+  protected long executeInsert(String sql, Object... params) {
+    try (Connection connection = getConnection()) {
+      connection.setAutoCommit(true);
+      PreparedStatement statement = connection.prepareStatement(sql);
+      try {
+        for (int i = 0, l = params.length; i < l; i++) {
+          statement.setObject(i + 1, params[i]);
+        }
+
+        statement.execute();
+        
+        try (ResultSet generatedKeys = statement.getGeneratedKeys()) {
+          if (generatedKeys.next()) {
+            return generatedKeys.getLong(1);
+          }
+        }
+      } finally {
+        statement.close();
+      }
+    } catch (Exception e) {
+      fail(e.getMessage());
+    }
+    
+    return -1;
+  }
+  
+  protected void executeDelete(String sql, Object... params) {
+    try (Connection connection = getConnection()) {
+      connection.setAutoCommit(true);
+      PreparedStatement statement = connection.prepareStatement(sql);
+      try {
+        for (int i = 0, l = params.length; i < l; i++) {
+          statement.setObject(i + 1, params[i]);
+        }
+
+        statement.execute();
+      } finally {
+        statement.close();
+      }
+    } catch (Exception e) {
+      fail(e.getMessage());
+    }
+  }
+
+  protected Connection getConnection() {
+    String username = System.getProperty("it.jdbc.username");
+    String password = System.getProperty("it.jdbc.password");
+    String url = System.getProperty("it.jdbc.url");
+    try {
+      Class.forName(System.getProperty("it.jdbc.driver")).newInstance();
+    } catch (InstantiationException | IllegalAccessException | ClassNotFoundException e) {
+      fail(e.getMessage());
+    }
+
+    try {
+      return DriverManager.getConnection(url, username, password);
+    } catch (SQLException e) {
+      fail(e.getMessage());
+    }
+    
+    return null;
+  }
+
+  protected static Matcher<Instant> sameInstant(final Instant instant) {
+    
+    return new BaseMatcher<Instant>(){
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce this anonymous class number of lines from 30 to at most 20, or make it a named class. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1188)